repo/content/content_manager_lock_free.go

```diff
@@ -31,7 +31,9 @@ func (sm *SharedManager) maybeCompressAndEncryptDataForPacking(data gather.Bytes
 
 	// If the content is prefixed (which represents Kopia's own metadata as opposed to user data),
 	// and we're on V2 format or greater, enable internal compression even when not requested.
-	if contentID.HasPrefix() && comp == NoCompression && mp.IndexVersion >= index.Version2 {
+	// Set compression only for manifest metadata.
+	// TODO: Remove this check once metadata compression setting is implemented for manifest metadata.
+	if contentID.HasPrefix() && contentID.Prefix() == ManifestContentPrefix && comp == NoCompression && mp.IndexVersion >= index.Version2 {
```

Comment:
can we lift this to the call site so that it's aways providing a compression choice?